### PR TITLE
Elewis clientimpl refactor

### PR DESCRIPTION
**Pull request by elewis787**
refactor of client impl 

next pass will include testing

### CODE REVIEW

File: clientimpl.go [hunk 1]
```diff
@@ -34,216 +32,239 @@ func NewMazzarothClient(servers []string, options ...Options) (*ClientImpl, erro
...
 	}
 
-	return response, nil
+	return nil, errors.New("Missing account")
 }
 
-// TransactionLookup calls the endpoint: /v1/channels/{channel_id}/transactions/{id}.
-func (c *ClientImpl) TransactionLookup(channelID string, transactionID string) (*xdr.Response, error) {
-	url := fmt.Sprintf("%s/%s/channels/%s/transactions/%s", c.serverSelector.Pick(), version, channelID, transactionID)
+// AuthroizationLookup calls the endpoing: /v1/channels/{channel_id}/accounts/{account_id}/authorized
```
Comment (jeff1010322) [inline clientimpl.go:57]: Spelling: "AuthorizationLookup calls the endpoint"

File: clientimpl.go [hunk 2]
```diff
@@ -34,216 +32,239 @@ func NewMazzarothClient(servers []string, options ...Options) (*ClientImpl, erro
...
+func (c *ClientImpl) AuthorizationLookup(ctx context.Context, channelID string, accountID string) (*xdr.Authorized, error) {
+	url := fmt.Sprintf("%s/%s/channels/%s/accounts/%s/authorized", c.address, version, channelID, accountID)
 
-	response, err := makeRequest(c.httpClient, http.MethodGet, url, nil)
+	xdrResp, err := c.do(ctx, url, http.MethodGet, nil)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to make a request to transaction lookup endpoint")
+		return nil, errors.Wrap(err, "uable to handle http response")
```
Comment (jeff1010322) [inline clientimpl.go:63]: Spelling: "unable"

File: clientimpl.go [hunk 3]
```diff
@@ -34,216 +32,239 @@ func NewMazzarothClient(servers []string, options ...Options) (*ClientImpl, erro
...
+		return nil, nil, errors.Wrap(err, "unable to make a request to transaction submit endpoint")
 	}
 
-	return response, nil
+	if xdrResp.TransactionID != nil {
+		return xdrResp.TransactionID, nil, nil
+	} else if xdrResp.Receipt != nil {
+		return &xdrResp.Receipt.TransactionID, xdrResp.Receipt, nil
+	}
+	return nil, nil, errors.New("unable to processes transaction")
```
Comment (jeff1010322) [inline clientimpl.go:224]: Should be "process"

File: clientimpl.go [hunk 4]
```diff
@@ -34,216 +32,239 @@ func NewMazzarothClient(servers []string, options ...Options) (*ClientImpl, erro
...
-		return nil, errors.Wrap(err, "unable to make a request to blockheaders lookup endpoint")
+		return nil, errors.Wrap(err, "unable to handle http response")
+	}
+
+	if xdrResp.BlockHeader != nil {
+		return xdrResp.BlockHeader, nil
 	}
 
-	return response, nil
+	return nil, errors.New("Mising blockheader")
```
Comment (jeff1010322) [inline clientimpl.go:133]: Spelling: "Missing"

File: clientimpl.go [hunk 5]
```diff
@@ -34,216 +32,239 @@ func NewMazzarothClient(servers []string, options ...Options) (*ClientImpl, erro
...
+
+	if xdrResp.Transaction != nil {
+		return xdrResp.Transaction, nil
+	}
+
+	return nil, errors.New("missing transaction")
+}
+
+func (c *ClientImpl) do(ctx context.Context, url string, method string, body []byte) (*xdr.Response, error) {
+	req, err := http.NewRequestWithContext(context.Background(), method, url, bytes.NewReader(body))
```
Comment (jeff1010322) [inline clientimpl.go:244]: Shouldn't this pass the function context here instead of creating a new one?

Comment (elewis787) [reply clientimpl.go:244]: yes thanks - copy past from the old method

**Review by jeff1010322 — APPROVED** [review]

**Pull request merged**